mne_connectivity/spectral/time.py

```diff
@@ -385,18 +512,28 @@ def spectral_connectivity_time(data, freqs, method='coh', average=False,
                                           conn_flat[m].shape[2:])
             conn[m] = this_conn
 
-    # create a Connectivity container
-    if average:
-        out = [SpectralConnectivity(
-               conn[m].mean(axis=0), freqs=out_freqs, n_nodes=n_signals,
-               names=names, indices=indices, method=method, spec_method=mode,
-               events=events, event_id=event_id, metadata=metadata)
-               for m in method]
-    else:
-        out = [EpochSpectralConnectivity(
-               conn[m], freqs=out_freqs, n_nodes=n_signals, names=names,
-               indices=indices, method=method, spec_method=mode, events=events,
-               event_id=event_id, metadata=metadata) for m in method]
+    if multivariate_con:
+        # UNTIL RAGGED ARRAYS SUPPORTED
+        indices = tuple(
+            [[np.array(indices_use[0])], [np.array(indices_use[1])]])
+
+    # create the connectivity containers
+    out = []
+    for m in method:
+        store_params = {
+            'data': conn[m], 'patterns': patterns[m], 'freqs': out_freqs,
+            'n_nodes': n_signals, 'names': names, 'indices': indices,
+            'method': method, 'spec_method': mode, 'events': events,
+            'event_id': event_id, 'metadata': metadata, 'rank': rank,
+            'n_lags': gc_n_lags if m in _gc_methods else None}
+        if average:
+            store_params['data'] = np.mean(store_params['data'], axis=0)
+            if patterns[m] is not None:
+                store_params['patterns'] = np.mean(store_params['patterns'],
```

Comment:
Can you add a unit-test for this case?

Reply:
Done! https://github.com/tsbinns/mne-connectivity/blob/894c42e59cef9f97aa07a7a4f71cef756512c50c/mne_connectivity/spectral/tests/test_spectral.py#L1050-L1088